ibis/backends/dask/execution/util.py

```diff
@@ -146,8 +143,15 @@ def coerce_to_output(
             divisions=(0, out_len - 1),
         )
         return series
+    elif isinstance(result, dd.DataFrame):
```

Comment:
I'm confused by the logic here - shouldn't any `dd.DataFrame` case be handled above? https://github.com/ibis-project/ibis/pull/5747/files#diff-7560a298b0242d66bc062059c8a1189235eea9088c8174ce22b7154caf1d629fR124